In [7]:
# Import the regular expression module
import re

# Compile the pattern: phone
phone = re.compile('\d{3}\-\d{3}\-\d{4}\-{1}')
# The above regular expression means the following 
# \d{3} : Start with a 3 digit number 
# \-    : A hypen (-) after the 3 digit number 
# \d{3} : Another 3 digit number after the previously found hypen(-)
# \-    : A hypen (-) after the 3 digit number 
# \d{4} : A 4 digit number followed by the previously found hypen(-) 

#“120/456-7890(8)”, 

result = phone.match("120/456-7890(8)")# Check if the pattern matches 
print("For 120/456-7890(8) :", bool(result))
#print(result.group())


For 120/456-7890(8) : False


In [9]:
#“123-456-789”, 
phone = re.compile('\d{3}\-\d{3}\-\d{3}')
result = phone.match("123-456-789")# Check if the pattern matches 
print(bool(result))
print(result.group())


True
123-456-789


In [10]:
#“120-456-7890(8)”. 
phone = re.compile('\d{3}\-\d{3}\-\d{3}\-d{1}')
result = phone.match("123-456-7890(8)")# Check if the pattern matches 
print(bool(result))
#print(result.group())


False


In [11]:
#TASK - 2                                   findall
import re                    # Importing re library 
file1 = open(r"C:\Users\Manish\Text paragraph.txt", "r").read()
print(file1)                 # To print contents of input file

Text Paragraph
A small paragraph is also provided as input for text
preprocessing and quick analysis.
A final note on ten key sentences is that academic
readers expect them to be at the beginning of the
paragraph. (The first sentence of this paragraph is one
good example of this!) This placement helps readers
comprehend your argument. To see how to try this: find
an academic piece (such as a textbook or scholarly
article) that strikes you like well-written and go through
part of it, just read the first sentence of each paragraph
You should be able to follow the sequence of logic.
When you’re writing for 22 professors, it is especially
effective to put your key sentences first to convey your original thinking


In [12]:
import re
mod_string = ''.join(filter(lambda item: not item.isdigit(), file1))         # Filter all digits from characters in string & join remaining character
print('\nAfter removal of Numeric values:\n\n',mod_string)                   # print text after removal of Numeric values


After removal of Numeric values:

 Text Paragraph
A small paragraph is also provided as input for text
preprocessing and quick analysis.
A final note on ten key sentences is that academic
readers expect them to be at the beginning of the
paragraph. (The first sentence of this paragraph is one
good example of this!) This placement helps readers
comprehend your argument. To see how to try this: find
an academic piece (such as a textbook or scholarly
article) that strikes you like well-written and go through
part of it, just read the first sentence of each paragraph
You should be able to follow the sequence of logic.
When you’re writing for  professors, it is especially
effective to put your key sentences first to convey your original thinking


In [13]:
item = mod_string                                                   # input file
text_new = re.sub(r'\s+',' ', item)                                 # Replacing extra white space by a single space
print('Input TEXT:\n\n',item)                                       # To print text before removing symbols
print('\n \n  After REMOVING extra White Spaces:\n\n',text_new)     # To print text after removing symbols

Input TEXT:

 Text Paragraph
A small paragraph is also provided as input for text
preprocessing and quick analysis.
A final note on ten key sentences is that academic
readers expect them to be at the beginning of the
paragraph. (The first sentence of this paragraph is one
good example of this!) This placement helps readers
comprehend your argument. To see how to try this: find
an academic piece (such as a textbook or scholarly
article) that strikes you like well-written and go through
part of it, just read the first sentence of each paragraph
You should be able to follow the sequence of logic.
When you’re writing for  professors, it is especially
effective to put your key sentences first to convey your original thinking

 
  After REMOVING extra White Spaces:

 Text Paragraph A small paragraph is also provided as input for text preprocessing and quick analysis. A final note on ten key sentences is that academic readers expect them to be at the beginning of the paragraph. (The first s

In [14]:
#TASK - 3

sample_string = text_new                                       # Taking text with removed whitespaces as input here                                                 
#print('Original text:',sample_string)                         # print original text
char_to_replace = {'!': '.'}                                 # Define characters to be replaced                                
for key, value in char_to_replace.items():                     # Iterate over all key-value pairs in dictionary
    sample_string = sample_string.replace(key, value)          # Replace key character with value character in string
print('After Replacing symbol:\n\n',sample_string)             # print text after replacing characters

After Replacing symbol:

 Text Paragraph A small paragraph is also provided as input for text preprocessing and quick analysis. A final note on ten key sentences is that academic readers expect them to be at the beginning of the paragraph. (The first sentence of this paragraph is one good example of this.) This placement helps readers comprehend your argument. To see how to try this: find an academic piece (such as a textbook or scholarly article) that strikes you like well-written and go through part of it, just read the first sentence of each paragraph You should be able to follow the sequence of logic. When you’re writing for professors, it is especially effective to put your key sentences first to convey your original thinking


In [34]:
#TASK - 4

import re
string = ('€','123','123.','123.4')
s= str(string)
if re.search('€',s):
    print("Match found")
else:
    print("No match found")

#import re
#string = ('€','123','123.','123.4')
#match = re.search('123', string)
#if match:
   # process(match)

Match found


In [35]:
#TASK - 5

import spacy                                  # Import spaCy library
nlp = spacy.load("en_core_web_sm")            # load spaCy model

In [36]:
file_name = '48_hrs.txt'                                      # File name
introduction_file_text = open(file_name).read()                 # To open text file
introduction_file_doc = nlp(introduction_file_text)             # Create NLP object
print(introduction_file_doc)                               # To print contents


					"48 HRS."

					   by

				   Roger Spottiswoode and
				   Walter Hill

					   &

				   Larry Gross and
				   Steven E. De Souza

FADE IN:

OPEN COUNTRY - DAY

	Endless green hills bisected by a ribbon of highway. A road
	gang clearing brush by the side of the road... Twenty-five
	men in prison fatigues sweating through their mid-afternoon
	labor.

	THREE GUARDS

	Flank the working prisoners... Mountie hats, shotguns,
	sidearms, sunglasses; they look like they mean it.

	HIGHWAY

	A battered pickup appears...approaches. Suddenly, it coughs,
	shudders, stalls. A big Blackfoot Indian named BILLY BEAR
	gets out and starts cursing and kicking the vehicle.Then he
	begins walking toward the road gang...

	ROADSIDE

	BRADY is the Guard near the center of the work gang; he
	smiles at the oncoming man, pokes a prisoner beside him.

					BRADY
			Wonder what reservation they let
			him off of...

	The prisoner is GANZ who looks up, grins at Brady...

					GANZ
			Yeah, there goes the n

In [37]:
nlp.pipe_names                                  # To display active pipeline components:

['tok2vec', 'tagger', 'parser', 'attribute_ruler', 'lemmatizer', 'ner']

In [39]:
nlp.disable_pipes('tagger', 'lemmatizer', 'ner', 'parser')          # to disable the pipeline components
nlp.pipe_names                                                      # Again checking active pipeline components

['tok2vec', 'attribute_ruler']

In [40]:
doc = nlp("Endless green hills bisected by a ribbon of highway.")     # Text as NLP object
print('Text1:',doc)                                                   # To print the text
doc2 = nlp("A road gang clearing brush by the side of the road... Twenty-five men in prison fatigues sweating through their mid-afternoon labor.")   # Text as NLP object
print('\nText2:',doc2)                                                # To print the text

Text1: Endless green hills bisected by a ribbon of highway.

Text2: A road gang clearing brush by the side of the road... Twenty-five men in prison fatigues sweating through their mid-afternoon labor.


In [41]:
print ([token.text for token in doc])        # Extract tokens for the given doc

['Endless', 'green', 'hills', 'bisected', 'by', 'a', 'ribbon', 'of', 'highway', '.']


### TASK - 6

In [47]:
from spacy.lang.en import English                              # Importing library
nlp = English()                                                # Importing model
nlp = spacy.load("en_core_web_sm")                             # Importing model
f = open("48_hrs.txt")                              #No need to specify the path speciy file name instead of path
contents = f.read()                                            # To read input dataset
contents = contents[:1500]                                      # First few characters of input file
print(contents)                                                # To print contents of file
text_combined = str(contents)                                  # String
doc = nlp(text_combined)                                       # Create NLP object
for token in doc:
    print(token)                                               # Print tokens
len(token)                                                     # Length of token


					"48 HRS."

					   by

				   Roger Spottiswoode and
				   Walter Hill

					   &

				   Larry Gross and
				   Steven E. De Souza

FADE IN:

OPEN COUNTRY - DAY

	Endless green hills bisected by a ribbon of highway. A road
	gang clearing brush by the side of the road... Twenty-five
	men in prison fatigues sweating through their mid-afternoon
	labor.

	THREE GUARDS

	Flank the working prisoners... Mountie hats, shotguns,
	sidearms, sunglasses; they look like they mean it.

	HIGHWAY

	A battered pickup appears...approaches. Suddenly, it coughs,
	shudders, stalls. A big Blackfoot Indian named BILLY BEAR
	gets out and starts cursing and kicking the vehicle.Then he
	begins walking toward the road gang...

	ROADSIDE

	BRADY is the Guard near the center of the work gang; he
	smiles at the oncoming man, pokes a prisoner beside him.

					BRADY
			Wonder what reservation they let
			him off of...

	The prisoner is GANZ who looks up, grins at Brady...

					GANZ
			Yeah, there goes the n

3

In [48]:
from collections import Counter
counts = Counter()
for token in doc:
    counts[token.orth_] += 1                       # Equivalently, token.text(orth means token.text)
print(counts)                                      # To print the frequency count

Counter({',': 16, '\n\t\t\t': 15, '.': 13, 'the': 12, '\n\n\t': 10, '...': 10, '\n\t': 8, '\n\n\t\t\t\t\t': 8, 'and': 5, 'of': 5, 'a': 4, 'BILLY': 4, 'GANZ': 4, 'by': 3, '-': 3, 'A': 3, 'road': 3, 'gang': 3, 'they': 3, 'BRADY': 3, 'him': 3, "'s": 3, '"': 2, '\n\n\t\t\t\t\t   ': 2, '\n\n\t\t\t\t   ': 2, '\n\t\t\t\t   ': 2, '\n\n': 2, 'in': 2, ';': 2, 'it': 2, 'big': 2, 'out': 2, 'he': 2, 'is': 2, 'at': 2, 'prisoner': 2, 'Brady': 2, 'as': 2, 'Billy': 2, 'on': 2, 'I': 2, 'got': 2, 'water': 2, '\n\t\t\t\t\t': 1, '48': 1, 'HRS': 1, 'Roger': 1, 'Spottiswoode': 1, 'Walter': 1, 'Hill': 1, '&': 1, 'Larry': 1, 'Gross': 1, 'Steven': 1, 'E.': 1, 'De': 1, 'Souza': 1, 'FADE': 1, 'IN': 1, ':': 1, 'OPEN': 1, 'COUNTRY': 1, 'DAY': 1, 'Endless': 1, 'green': 1, 'hills': 1, 'bisected': 1, 'ribbon': 1, 'highway': 1, 'clearing': 1, 'brush': 1, 'side': 1, 'Twenty': 1, 'five': 1, 'men': 1, 'prison': 1, 'fatigues': 1, 'sweating': 1, 'through': 1, 'their': 1, 'mid': 1, 'afternoon': 1, 'labor': 1, 'THREE': 1, 'GU

### TASK - 7

In [69]:

#John went to U.S.A with his elder sister
import spacy
from spacy.lang.en import English
nlp = English()                                                # Importing model
nlp = spacy.load("en_core_web_sm")
doc = nlp("John went to U.S.A with his elder sister")     # Text as NLP object
print('Text1:',doc)                           # To print the text
#print ([token.text for token in doc])        # Extract tokens for the given doc

Text1: John went to U.S.A with his elder sister


In [70]:
print ([token.text for token in doc])        # Extract tokens for the given doc

['John', 'went', 'to', 'U.S.A', 'with', 'his', 'elder', 'sister']


U.S.A be considered as one token and total token count is 8.

### TASK - 8


In [29]:
import string
text = open("48_hrs.txt","r")
d= dict()
for line in text:
    line = line.strip()
    line= line.lower()
    line=line.translate(line.maketrans("", "",string.punctuation))
    words = line.split(" ")
    for word in words:
        if word in d:
            d[word] = d[word]+1
        else:
            d[word] = 1
for key in list(d.keys()):
    print(key, ":", d[key])

 : 2334
48 : 3
hrs : 1
by : 45
roger : 1
spottiswoode : 1
and : 306
walter : 1
hill : 4
larry : 1
gross : 1
steven : 1
e : 1
de : 2
souza : 1
fade : 1
in : 207
open : 22
country : 2
day : 14
endless : 1
green : 5
hills : 1
bisected : 1
a : 603
ribbon : 1
of : 268
highway : 4
road : 9
gang : 9
clearing : 1
brush : 1
the : 1071
side : 26
twentyfive : 1
men : 14
prison : 12
fatigues : 2
sweating : 1
through : 30
their : 14
midafternoon : 1
labor : 1
three : 19
guards : 5
flank : 1
working : 3
prisoners : 3
mountie : 1
hats : 1
shotguns : 1
sidearms : 1
sunglasses : 2
they : 100
look : 46
like : 62
mean : 8
it : 219
battered : 3
pickup : 3
appearsapproaches : 1
suddenly : 15
coughs : 1
shudders : 1
stalls : 1
big : 30
blackfoot : 1
indian : 21
named : 6
billy : 87
bear : 15
gets : 25
out : 131
starts : 33
cursing : 1
kicking : 1
vehiclethen : 1
he : 192
begins : 14
walking : 3
toward : 36
roadside : 2
brady : 9
is : 127
guard : 23
near : 14
center : 1
work : 13
smiles : 41
at : 169
oncomin

In [30]:
import string
text = open("scifiscripts.txt","r")
d= dict()
for line in text:
    line = line.strip()
    line= line.lower()
    line=line.translate(line.maketrans("", "",string.punctuation))
    words = line.split(" ")
    for word in words:
        if word in d:
            d[word] = d[word]+1
        else:
            d[word] = 1
for key in list(d.keys()):
    print(key, ":", d[key])

note : 3
from : 66
poster : 1
to : 370
kubrick : 3
newsgroup : 1
 : 1689
i : 162
found : 7
this : 102
on : 106
a : 268
bbs : 1
while : 5
ago : 10
and : 382
thought : 14
id : 10
pass : 3
it : 199
along : 4
all : 57
of : 401
you : 178
freaks : 1
out : 60
there : 61
022389 : 1
transcribers : 1
for : 111
clarkekubrick2001 : 1
fans : 1
the : 867
original : 5
paper : 2
copy : 4
screenplay : 5
back : 21
felt : 6
compelled : 1
transcribe : 1
disk : 1
upload : 1
various : 4
bulletin : 1
boards : 1
enjoyment : 1
final : 2
movie : 1
deviates : 1
in : 199
number : 8
interesting : 2
ways : 1
ive : 17
tried : 1
maintain : 2
format : 1
document : 1
except : 4
lines : 2
per : 4
page : 2
order : 6
reduce : 1
length : 1
file : 2
used : 4
bar : 4
delimit : 1
pages : 3
as : 67
was : 82
lot : 4
whitespace : 1
2001 : 4
space : 42
odyssey : 1
by : 48
stanley : 1
arthur : 1
c : 1
clark : 1
hawk : 1
films : 2
ltd : 1
co : 1
mgm : 1
studios : 1
boreham : 1
wood : 1
herts : 1
title					 : 1
part : 8
africa : 3
3

In [32]:
import string
text = open("Ten_things_I_hate.txt","r")
d= dict()
for line in text:
    line = line.strip()
    line= line.lower()
    line=line.translate(line.maketrans("", "",string.punctuation))
    words = line.split(" ")
    for word in words:
        if word in d:
            d[word] = d[word]+1
        else:
            d[word] = 1
for key in list(d.keys()):
    print(key, ":", d[key])

ten : 2
things : 5
i : 299
hate : 18
about : 41
you : 387
 : 2612
written : 1
by : 46
karen : 1
mccullah : 1
lutz : 1
kirsten : 1
smith : 1
based : 1
on : 124
taming : 2
of : 210
the : 687
shrew : 3
william : 10
shakespeare : 8
revision : 1
november : 1
12 : 1
1997 : 1
padua : 6
high : 10
school : 29
day : 72
welcome : 2
to : 444
your : 69
typical : 1
urbansuburban : 1
in : 201
portland : 1
oregon : 1
smarties : 1
skids : 1
preppies : 1
granolas : 1
loners : 1
lovers : 1
and : 385
out : 111
crowd : 12
rub : 1
sleep : 6
their : 28
eyes : 15
head : 25
for : 95
main : 1
building : 2
parking : 7
lot : 11
kat : 409
stratford : 22
eighteen : 2
pretty : 8
but : 65
trying : 5
hard : 3
not : 77
be : 47
a : 477
baggy : 1
granny : 1
dress : 11
glasses : 7
balances : 1
cup : 3
coffee : 10
backpack : 2
as : 111
she : 167
climbs : 2
her : 297
battered : 2
baby : 2
blue : 3
75 : 1
dodge : 1
dart : 2
stray : 3
skateboard : 3
clips : 1
causing : 1
stumble : 1
spill : 1
well : 21
contents : 1
young : 1


Count of word "The" in '48_hrs' file : 1071

Count of word "The" in 'scifiscripts file' : 867 

Count of word "The" in 'Ten_things_I_hate' file : 687
    
So, the file 'Ten_things_I_hate' has the least number of count of word "The".    